x-pack/plugins/security_solution/public/detections/pages/detection_engine/rules/all/batch_actions.tsx

```diff
@@ -51,11 +54,27 @@ export const getBatchItems = ({
     (id) => rules.find((r) => r.id === id)?.immutable ?? false
   );
 
+  const canBulkActivateRulesWithActions = selectedRuleIds.every((id) => {
+    const found = rules.find((rule) => rule.id === id);
+    if (found != null) {
+      if (found.actions.length > 0) {
+        if (canEditRuleWithActions(found, hasActionsPrivileges)) {
+          return true;
+        } else {
+          return false;
+        }
+      } else {
+        return true;
+      }
+    }
+    return true;
+  });
```

Comment:
This is a bit difficult to follow -- to determine the value of `canBulkActivateRulesWithActions`, we really just need to know if the user `hasActionsPrivileges`, and if _any_ of the selected rules have actions, right?

With this logic we're going to loop over every rule even if the first rule has actions and the user doesn't have permissions, even though that's enough to know that we need to disable any bulk_action that requires action privileges.

Still making use of your `canEditRuleWithActions()` function, we can invert the check using `some()` such that we'll bail early as soon as we find one rule with actions and the user doesn't `hasActionsPrivileges`. 

``` ts
  const missingActionPrivileges = selectedRuleIds.some((id) => {
    return !canEditRuleWithActions(
      rules.find((rule) => rule.id === id),
      hasActionsPrivileges
    );
  });
```

Going further, if the user `hasActionsPrivileges` we don't even have to check if any rule has actions, so we can short circuit even earlier! At the expense of readability of course... 😅 

``` ts
  const missingActionPrivileges =
    !hasActionsPrivileges &&
    selectedRuleIds.some((id) => {
      return !canEditRuleWithActions(
        rules.find((rule) => rule.id === id),
        hasActionsPrivileges
      );
    });
```

And then you can replace `!canBulkActivateRulesWithActions` with `missingActionPrivileges`, which reads a bit easier too since it [avoids negation](https://schneide.blog/2014/08/03/dont-ever-not-avoid-negative-logic/) (although it adds one back into the calculation, but that can't be avoided since we're inverting our check).